system_tests/common_test.go

```diff
@@ -60,6 +60,182 @@ import (
 type info = *BlockchainTestInfo
 type client = arbutil.L1Interface
 
+type TestClient struct {
+	ctx     context.Context
+	Info    info
+	Client  *ethclient.Client
+	Backend *eth.Ethereum
+	Stack   *node.Node
+	Node    *arbnode.Node
+}
+
+func NewTestClient(ctx context.Context) *TestClient {
+	return &TestClient{ctx: ctx}
+}
+
+func (tc *TestClient) SendSignedTx(t *testing.T, l2Client *ethclient.Client, transaction *types.Transaction) *types.Receipt {
+	return SendSignedTxViaL1(t, tc.ctx, tc.Info, tc.Client, l2Client, transaction)
+}
+
+func (tc *TestClient) SendUnsignedTx(t *testing.T, l2Client *ethclient.Client, transaction *types.Transaction) *types.Receipt {
+	return SendUnsignedTxViaL1(t, tc.ctx, tc.Info, tc.Client, l2Client, transaction)
+}
+
+func (tc *TestClient) TransferBalance(t *testing.T, from string, to string, amount *big.Int) (*types.Transaction, *types.Receipt) {
+	return TransferBalanceTo(t, from, tc.Info.GetAddress(to), amount, tc.Info, tc.Client, tc.ctx)
+}
+
+func (tc *TestClient) TransferBalanceTo(t *testing.T, from string, to common.Address, amount *big.Int) (*types.Transaction, *types.Receipt) {
+	return TransferBalanceTo(t, from, to, amount, tc.Info, tc.Client, tc.ctx)
+}
+
+func (tc *TestClient) GetBalance(t *testing.T, account common.Address) *big.Int {
+	return GetBalance(t, tc.ctx, tc.Client, account)
+}
+
+func (tc *TestClient) GetBaseFeeAt(t *testing.T, blockNum *big.Int) *big.Int {
+	return GetBaseFeeAt(t, tc.Client, tc.ctx, blockNum)
+}
+
+func (tc *TestClient) SendWaitTestTransactions(t *testing.T, txs []*types.Transaction) {
+	SendWaitTestTransactions(t, tc.ctx, tc.Client, txs)
+}
+
+func (tc *TestClient) DeploySimple(t *testing.T, auth bind.TransactOpts) (common.Address, *mocksgen.Simple) {
+	return deploySimple(t, tc.ctx, auth, tc.Client)
+}
+
+type NodeBuilder struct {
+	// NodeBuilder configuration
+	ctx           context.Context
+	Info          info
+	chainConfig   *params.ChainConfig
+	nodeConfig    *arbnode.Config
+	stackConfig   *node.Config
+	cachingConfig *execution.CachingConfig
+
+	// L1, L2 Node parameters
+	isSequencer   bool
+	takeOwnership bool
+	withL1        bool
+
+	// Blockchain parameters
+	dataDir     string
+	initMessage *arbostypes.ParsedInitMessage
+
+	// Created nodes
+	L1 *TestClient
+	L2 *TestClient
+}
+
+func NewNodeBuilder(ctx context.Context) *NodeBuilder {
+	return &NodeBuilder{ctx: ctx}
+}
+
+func (b *NodeBuilder) DefaultConfig(withL1 bool, n *arbnode.Config, c *params.ChainConfig) *NodeBuilder {
```

Comment:
i'd go even further and say you don't need arbnode.Config or chainConfig here.. let test change them after calling this if it wants.

Reply:
likewise - this should set all configs to default, including stackconfig (I think separate for l1/l2), default l1/l2info, default chainconfig.. none of the fuctions called later in the build process should get nil and use default for anything.